test/E2E/E2ETest.cs

```diff
@@ -50,6 +51,78 @@ public void TestDotnetCompile()
             TestOutputExecutable(OutputDirectory);
         }
 
+        [Fact] 
+        public void TestDotnetIncrementalBuild()
+        {
+            TestSetup();
+
+            //first build
+            TestRunCommand("dotnet", $" --verbose build --build-profile -o {OutputDirectory}");
```

Comment:
This should not be too bad, but I realized the --build-profile flag will need to be added to BuildCommand.cs. All other flags should be there:

https://github.com/dotnet/cli/blob/master/test/Microsoft.DotNet.Tools.Tests.Utilities/Commands/BuildCommand.cs